File: App/app/src/main/java/com/github/travelplannerapp/login/LoginActivity.java
package com.github.travelplannerapp.login;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.Button;

import com.github.travelplannerapp.travels.TravelsActivity;
import com.github.travelplannerapp.R;

public class LoginActivity extends AppCompatActivity implements LoginContract.View {

    LoginContract.Presenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        //set up presenter
        presenter =  new LoginPresenter(this);

        //set up toolbar
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        //set up button
        Button button = findViewById(R.id.button);
        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                presenter.signIn();
            }
        });
    }

    @Override
    public void showTravels() {
        Intent intent = new Intent(this, TravelsActivity.class);
        startActivity(intent);
    }
}

File: App/app/src/main/java/com/github/travelplannerapp/login/LoginActivity.kt
package com.github.travelplannerapp.login

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.Toolbar
import android.view.View
import android.widget.Button

import com.github.travelplannerapp.travels.TravelsActivity
import com.github.travelplannerapp.R

import javax.inject.Inject

import dagger.android.AndroidInjection

class LoginActivity : AppCompatActivity(), LoginContract.View {

    @Inject
    lateinit var presenter: LoginPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        //set up toolbar
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        //set up button
        val button = findViewById<Button>(R.id.button)
        button.setOnClickListener { presenter.signIn() }
    }

    override fun showTravels() {
        val intent = Intent(this, TravelsActivity::class.java)
        startActivity(intent)
    }
}
